include/boost_plugin_loader/plugin_loader.hpp

```diff
@@ -19,15 +19,44 @@
 #ifndef BOOST_PLUGIN_LOADER_PLUGIN_LOADER_HPP
 #define BOOST_PLUGIN_LOADER_PLUGIN_LOADER_HPP
 
+// STD
+#include <sstream>
+
+// Boost
 #include <boost/core/demangle.hpp>
 #include <boost/dll/import.hpp>
-#include <sstream>
+#include <boost/dll/runtime_symbol_info.hpp>
+#include <boost/version.hpp>
 
+// Boost Plugin Loader
 #include <boost_plugin_loader/plugin_loader.h>
 #include <boost_plugin_loader/utils.h>
 
 namespace boost_plugin_loader
 {
+/**
+ * @brief Separates a set of library paths with mixed path specification into one set with libraries that are specified
+ * as fully-defined absolute paths and a second set with libraries that are specified by name only.
+ * @param library_names The set to search and remove libraries with full paths
+ * @return A array containing 1) the set of the libraries specified as fully-defined absolute paths, and 2) the set of
+ * remaining libraries specified as library names
+ */
+inline std::array<std::set<std::string>, 2>
```

Comment:
Why an array instead of a tuple or pair? Seems like one of those data types would be more fitting

Reply:
The Cpp Core Guidelines use to recommend using an array over pair or tuple when the data is homogeneous. There is more overhead with the pair or tuple because it must support multiple data types. Also when the data is homogeneous it add extra ambiguity because it is not clear that the data is homogeneous where with a an array it is. The new Cpp Core Guidelines actual recommend to avoid using both pair and tuple and instead use a struct with clearly defined variables.

Reply:
Switched to tuple to leverage std::tie